docs/UserGuide.md

```diff
@@ -66,6 +66,118 @@ Shows a message explaning how to access the help page.
 
 Format: `help`
 
+### Delete a Task : `delete` 
+
+Delete a Task together with all its data, specified by its index.
+
+Format : `deletetask i/INDEX`
+
+- The index must be a **positive integer** and must be **valid**
+
+Example : 
+*  `deletetask i/1`
+
+### Create a Preset Task Template : `maketemplate`
+
+Create a template for recurring weekly tasks (Preset Tasks). It is initialized
+with no Preset Tasks.
+
+Format : `maketemplate n/TEMPLATENAME`
+
+### Add a Preset Task to a Template : `addpresettask` 
+
+Add a Preset Task to a Task Template, specified by its index.
+
+Format: `addpresettask i/INDEX n/TASKNAME sd/DATE d/DURATION`
+
+- `DURATION` is measured in hours
+- Addition of a Preset Task to a specific day will be **rejected** if the total tasks
+  for that day exceed 24 hours
+- The index of the template must be a **positive integer** and must be **valid**
+
+Example: 
+* `addpresettask i/1 n/Read Book sd/29-09-2020 d/2`
+
+### List Preset Task Templates : `listtemplates`
+
+List all Preset Task Templates.
+
+Format : `listtemplates`
+
+### List Preset Tasks : `listpresettasks`
+
+List all items in a Task Template, specified by its index.
+
+Format :  `listpresettasks i/INDEX`
+- The index must be a **positive integer** and must be **valid**
+
+Example
+* `listpresettasks i/1`
+
+### Load Template into Task Schedule : `loadtemplate` 
+
+Load a Preset Task Template into the Task Schedule, specified by its index.
```

Comment:
For now maybe we should use remove the spacing between Task Template, Task Schedule, Module Schedule etc.